torch/csrc/Module.cpp

```diff
@@ -537,23 +537,23 @@ PyObject *THPModule_hasDistributed(PyObject *_unused)
 #endif
 }
 
-void destroy_DLPack_PyCapsule(PyObject * obj) {
-  delete (DLTensor*)PyCapsule_GetPointer(obj, "tensor");
+void DLPackPyCapsule_destroy(PyObject * obj) {
+  DLManagedTensor * dlMTensor = (DLManagedTensor*)PyCapsule_GetPointer(obj, "dltensor");
```

Comment:
Recap of in-person discussion: PyCapsules holding DLManagedTensors should not do anything on deletion. They are just a container to transfer ownership of the DLManagedTensor from one framework to another. The expectation is that once you call `to_dlpack` the caller becomes responsible for ensuring that the deletor on the returned tensor gets called precisely once after it done with its use. That can be done by passing the DLManagedTensor to something that steals ownership (e.g. a  `from_dlpack` call) or by manually calling it. Because there is this manual behavior we will make to/from_dlpack a utility function and expect that higher-level functions public like to_caffe2 and from_pytorch compose them correctly.

Reply:
Hmmm ownership transfer is common in C/C++ interfaces, but I can't think of any Python APIs that work this way. Let's not increase mental overhead for users. I can see people blowing up programs just because they forgot they used a descriptor once, and copying them has almost zero cost.